ttnn/cpp/ttnn/operations/matmul/device/matmul_op.cpp

```diff
@@ -1022,16 +1025,44 @@ Matmul create_matmul_struct(
     bool broadcast_batch =
         parameters.bcast_batch.value_or(get_broadcast_batch(input_tensor_a, input_tensor_b, parameters.program_config));
     TT_FATAL(!(has_user_grid && has_program_config), "Cannot use both user core grid/coordinates and a program config");
+
+    const bool is_optional_output_tensor =
+        !optional_output_tensors.empty() && optional_output_tensors.at(0).has_value();
+    auto output_dtype = parameters.output_dtype;
```

Comment:
Pleaes don't use auto here for output_dtype and output_mem_config, especially since it looks like you're using optional here and just using it directly at the end of the method when returning the Matmul struct.